### PR TITLE
b_u585i_iot02a: Enable I2C, HTS221 sensor and stop modes

### PR DESCRIPTION
**Pull request by erwango**
This PR enables  I2C, HTS221 sensor and stop modes on b_u585i_iot02a

To support low power use cases:
- STM32 LPTIM driver is updated for U5 support
- A dedicated U5 power.c file is provided

### CODE REVIEW

**Review by ABOSTM — APPROVED** [review]
LGTM

File: dts/arm/st/u5/stm32u5.dtsi [hunk 1]
```diff
@@ -274,6 +274,18 @@
...
 			label= "I2C_2";
 		};
+
+		lptim1: timers@46004400 {
+			compatible = "st,stm32-lptim";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x46004400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB3 0x00000800>;
+			interrupts = <67 0>;
```
Comment (FRASTM) [inline dts/arm/st/u5/stm32u5.dtsi:284]: should the priority set =1 (becasue of https://github.com/zephyrproject-rtos/zephyr/pull/38066)

Comment (erwango) [reply dts/arm/st/u5/stm32u5.dtsi]: @FRASTM Good catch, fixed, txs

File: boards/arm/b_u585i_iot02a/doc/index.rst [hunk 2]
```diff
@@ -175,7 +175,7 @@ The Zephyr b_u585i_iot02a board configuration supports the following hardware fe
 +-----------+------------+-------------------------------------+
 | GPIO      | on-chip    | gpio                                |
 +-----------+------------+-------------------------------------+
-| I2C       | on-chip    | i2c                                |
+| I2C       | on-chip    | i2c                                 |
```
Comment (MaureenHelm) [inline boards/arm/b_u585i_iot02a/doc/index.rst]: This change needs to move to commit `boards: b_u585i_iot02a: Enable i2c and hts221 mems sensors`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: boards/arm/b_u585i_iot02a/doc/index.rst (hunk 2 above)
Comment (erwango) [reply boards/arm/b_u585i_iot02a/doc/index.rst]: Good catch, txs

**Review by tejlmand — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Comment by MaureenHelm** [comment]
Needs rebase

**Comment by ycsin** [comment]
@cfriedt @erwango I think new hardware support should be backported?

**Comment by erwango** [comment]
@ycsin this cannot be backported as these are not bug fixes

**Pull request merged**